File: db/typescript-mysql/src/api/domain/product.ts
// src/domain/product.ts
import { prisma } from '../lib/prisma'
import { Product } from '@prisma/client'

export interface CreateProductData {
  prodCode: string
  fullname: string
  name: string
  kana: string
  unitprice: number
  primeCost: number
  supCode: string
}

export interface UpdateProductData {
  fullname?: string
  name?: string
  kana?: string
  unitprice?: number
  primeCost?: number
  supCode?: string
}

export class ProductDomain {
  /**
   * 商品を作成
   */
  async create(data: CreateProductData): Promise<Product> {
    return await prisma.product.create({
      data: {
        ...data,
        taxType: 1 // デフォルト値
      }
    })
  }

  /**
   * すべての商品を取得
   */
  async findAll(): Promise<Product[]> {
    return await prisma.product.findMany()
  }

  /**
   * ID で商品を取得
   */
  async findById(prodCode: string): Promise<Product | null> {
    return await prisma.product.findUnique({
      where: { prodCode }
    })
  }

  /**
   * 商品を更新
   */
  async update(prodCode: string, data: UpdateProductData): Promise<Product> {
    return await prisma.product.update({
      where: { prodCode },
      data
    })
  }

  /**
   * 商品を削除
   */
  async delete(prodCode: string): Promise<void> {
    await prisma.product.delete({
      where: { prodCode }
    })
  }
}

File: db/go/sales_management/internal/domain/product.go
package domain

import "time"

// Product は API レイヤー用の商品ドメインモデルです
type Product struct {
	ProdCode     string    `db:"商品コード"`
	FullName     string    `db:"商品正式名"`
	Name         string    `db:"商品略称"`
	Kana         *string   `db:"商品名カナ"`
	UnitPrice    int       `db:"販売単価"`
	PoPrice      int       `db:"仕入単価"`
	SupCode      *string   `db:"仕入先コード"`
	CategoryCode *string   `db:"商品分類コード"`
	CreateDate   time.Time `db:"作成日時"`
	Creator      *string   `db:"作成者名"`
	UpdateDate   time.Time `db:"更新日時"`
	Updater      *string   `db:"更新者名"`
}
